lib/main.dart

```diff
@@ -50,53 +50,108 @@ class App extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    // Initialize configuration and service repository.
+    return _WithServiceRepository(
+      child: _WithSelectedNetwork(
+        initialNetwork: initialNetwork,
+        child: MultiBlocProvider(
+          providers: [
+            BlocProvider(
+              create: (context) {
+                // Initialize T&C by loading the currently accepted version from shared preferences.
+                final prefs = context.read<ServiceRepository>().sharedPreferences;
+                return TermsAndConditionAcceptance(prefs);
+              },
+            ),
+          ],
+          child: MaterialApp(
+            routes: appRoutes,
+            theme: concordiumTheme(),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _WithServiceRepository extends StatefulWidget {
+  final Widget child;
+
+  const _WithServiceRepository({super.key, required this.child});
+
+  @override
+  State<_WithServiceRepository> createState() => _WithServiceRepositoryState();
+}
+
+class _WithServiceRepositoryState extends State<_WithServiceRepository> {
+  Future<ServiceRepository>? _future;
```

Comment:
```suggestion
  late final Future<ServiceRepository> _future;
```

Reply:
Why this change? `FutureBuilder` accepts the value as nullable so this seemed natural to me.

I'm confused about the meaning of `late` and `final` in the same declaration. They seem kind of contradictory to me :thinking:

Reply:
`final` means it can only be set once, and `late` means it doesn't have to happen when the object is created. But it does read a little weird to me as well :smile: 

It will throw an exception if you try to assign `_future` more than once when `late final`.

Reply:
Yeah just tried it out and that's also what I observed. I think it makes sense; it should never be assigned more than once so getting an error (`LateInitializationError`) in that case seems like the right behavior.